code/policies/BackwardsCompatibleCachingPolicy.php

```diff
@@ -49,113 +49,113 @@ class BackwardsCompatibleCachingPolicy extends HTTP implements ControllerPolicy
+            HTTPCacheControl::singleton()->disableCaching();
         }
 
         // Populate $responseHeaders with all the headers that we want to build
         $responseHeaders = array();
-        if (function_exists('apache_request_headers')) {
-            $requestHeaders = apache_request_headers();
-            if (isset($requestHeaders['X-Requested-With']) && $requestHeaders['X-Requested-With']=='XMLHttpRequest') {
-                $cacheAge = 0;
-            }
-            // bdc: now we must check for DUMB IE6:
-            if (isset($requestHeaders['x-requested-with']) && $requestHeaders['x-requested-with']=='XMLHttpRequest') {
-                $cacheAge = 0;
-            }
-        }
+        $cacheControlHeaders = $config->get('cache_control');
+        if (!$config->get('cache_ajax_requests') && function_exists('apache_request_headers')) {
+            $requestHeaders = array_change_key_case(apache_request_headers(), CASE_LOWER);
 
-        if ($cacheAge > 0) {
-            $responseHeaders["Cache-Control"] = "max-age=" . $cacheAge . ", must-revalidate, no-transform";
-
-            // Set empty pragma to avoid PHP's session_cache_limiter adding conflicting caching information,
-            // defaulting to "nocache" on most PHP configurations (see http://php.net/session_cache_limiter).
-            // Since it's a deprecated HTTP 1.0 option, all modern HTTP clients and proxies should
-            // prefer the caching information indicated through the "Cache-Control" header.
-            $responseHeaders["Pragma"] = "";
-
-            $responseHeaders['Vary'] = $this->vary;
-        } else {
-            if ($response) {
-                // Grab header for checking. Unfortunately HTTPRequest until 3.1 uses a mistyped variant.
-                $contentDisposition = $response->getHeader('Content-disposition');
-                if (!$contentDisposition) {
-                    $contentDisposition = $response->getHeader('Content-Disposition');
-                }
+            if (array_key_exists('x-requested-with', $requestHeaders) && strtolower($requestHeaders['x-requested-with']) == 'xmlhttprequest') {
+                HTTPCacheControl::singleton()->disableCaching();
             }
+        }
 
-            if (
-                $response &&
-                Director::is_https() &&
-                strstr($_SERVER["HTTP_USER_AGENT"], 'MSIE')==true &&
-                strstr($contentDisposition, 'attachment;')==true
-            ) {
-                // IE6-IE8 have problems saving files when https and no-cache are used
-                // (http://support.microsoft.com/kb/323308)
-                // Note: this is also fixable by ticking "Do not save encrypted pages to disk" in advanced options.
-                $responseHeaders["Cache-Control"] = "max-age=3, must-revalidate, no-transform";
-                $responseHeaders["Pragma"] = "";
+        $vary = $config->get('vary');
+        if ($vary && strlen($vary)) {
```

Comment:
But: Since `BackwardsCompatibleCachingPolicy` is removed from CWP 2.0, the duplication will disappear as well, with those lines only showing in the `CachingPolicy` implementation further down. Might not be worth the trouble.

Reply:
This could be part of an `addVary` method if we go down that route, as a first run I'm just looking at upgrading what's there to be more consistent with framework